### PR TITLE
Explicit call `__construct()` method in `DebugBlocksCommand`

### LINKED ISSUE #624: Some commands could not be registered
#### Sonata packages

```bash
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.52.0 3.52.0 The missing Symfony Admin Generator
sonata-project/block-bundle              3.17.0 3.17.0 Symfony SonataBlockBundle
sonata-project/cache                     2.0.1  2.0.1  Cache library
sonata-project/core-bundle               3.17.0 3.17.0 Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.5.0  3.0.0  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.3.0  1.3.0  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.10.0 3.10.0 Symfony Sonata / Integrate Doctrine ORM into the SonataAdm...
sonata-project/easy-extends-bundle       2.5.0  2.5.0  Symfony SonataEasyExtendsBundle
sonata-project/exporter                  2.0.1  2.0.1  Lightweight Exporter library
sonata-project/user-bundle               4.4.0  4.4.0  Symfony SonataUserBundle
```

#### Symfony packages

```bash
$ composer show --latest 'symfony/*'
symfony/asset                      v4.3.3  v4.3.3  Symfony Asset Component
symfony/browser-kit                v4.3.3  v4.3.3  Symfony BrowserKit Component
symfony/cache                      v4.3.3  v4.3.3  Symfony Cache component with PSR-6, PSR-16, and tags
symfony/cache-contracts            v1.1.5  v1.1.5  Generic abstractions related to caching
symfony/class-loader               v3.4.30 v3.4.30 Symfony ClassLoader Component
symfony/config                     v4.3.3  v4.3.3  Symfony Config Component
symfony/console                    v4.3.3  v4.3.3  Symfony Console Component
symfony/css-selector               v3.4.30 v4.3.3  Symfony CssSelector Component
symfony/debug                      v4.3.3  v4.3.3  Symfony Debug Component
symfony/debug-bundle               v4.3.3  v4.3.3  Symfony DebugBundle
symfony/debug-pack                 v1.0.7  v1.0.7  A debug pack for Symfony projects
symfony/dependency-injection       v4.3.3  v4.3.3  Symfony DependencyInjection Component
symfony/doctrine-bridge            v4.3.3  v4.3.3  Symfony Doctrine Bridge
symfony/dom-crawler                v4.3.3  v4.3.3  Symfony DomCrawler Component
symfony/dotenv                     v4.3.3  v4.3.3  Registers environment variables from a .env file
symfony/event-dispatcher           v4.3.3  v4.3.3  Symfony EventDispatcher Component
symfony/event-dispatcher-contracts v1.1.5  v1.1.5  Generic abstractions related to dispatching event
symfony/expression-language        v4.3.3  v4.3.3  Symfony ExpressionLanguage Component
symfony/filesystem                 v4.3.3  v4.3.3  Symfony Filesystem Component
symfony/finder                     v4.3.3  v4.3.3  Symfony Finder Component
symfony/flex                       v1.4.5  v1.4.5  Composer plugin for Symfony
symfony/form                       v4.3.3  v4.3.3  Symfony Form Component
symfony/framework-bundle           v4.3.3  v4.3.3  Symfony FrameworkBundle
symfony/http-foundation            v4.3.3  v4.3.3  Symfony HttpFoundation Component
symfony/http-kernel                v4.3.3  v4.3.3  Symfony HttpKernel Component
symfony/inflector                  v4.3.3  v4.3.3  Symfony Inflector Component
symfony/intl                       v4.3.3  v4.3.3  A PHP replacement layer for the C intl extension that includes...
symfony/lock                       v4.3.3  v4.3.3  Symfony Lock Component
symfony/mailer                     v4.3.3  v4.3.3  Symfony Mailer Component
symfony/maker-bundle               v1.13.0 v1.13.0 Symfony Maker helps you create empty commands, controllers, fo...
symfony/mime                       v4.3.3  v4.3.3  A library to manipulate MIME messages
symfony/monolog-bridge             v4.3.3  v4.3.3  Symfony Monolog Bridge
symfony/monolog-bundle             v3.4.0  v3.4.0  Symfony MonologBundle
symfony/options-resolver           v4.3.3  v4.3.3  Symfony OptionsResolver Component
symfony/orm-pack                   v1.0.6  v1.0.6  A pack for the Doctrine ORM
symfony/phpunit-bridge             v4.3.3  v4.3.3  Symfony PHPUnit Bridge
symfony/polyfill-intl-icu          v1.11.0 v1.12.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn          v1.12.0 v1.12.0 Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 funct...
symfony/polyfill-mbstring          v1.12.0 v1.12.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72             v1.12.0 v1.12.0 Symfony polyfill backporting some PHP 7.2+ features to lower P...
symfony/polyfill-php73             v1.12.0 v1.12.0 Symfony polyfill backporting some PHP 7.3+ features to lower P...
symfony/process                    v4.3.3  v4.3.3  Symfony Process Component
symfony/profiler-pack              v1.0.4  v1.0.4  A pack for the Symfony web profiler
symfony/property-access            v4.3.3  v4.3.3  Symfony PropertyAccess Component
symfony/property-info              v4.3.3  v4.3.3  Symfony Property Info Component
symfony/routing                    v4.3.3  v4.3.3  Symfony Routing Component
symfony/security-acl               v3.0.2  v3.0.2  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle            v4.3.3  v4.3.3  Symfony SecurityBundle
symfony/security-core              v4.3.3  v4.3.3  Symfony Security Component - Core Library
symfony/security-csrf              v4.3.3  v4.3.3  Symfony Security Component - CSRF Library
symfony/security-guard             v4.3.3  v4.3.3  Symfony Security Component - Guard
symfony/security-http              v4.3.3  v4.3.3  Symfony Security Component - HTTP Integration
symfony/serializer                 v4.3.3  v4.3.3  Symfony Serializer Component
symfony/serializer-pack            v1.0.2  v1.0.2  A pack for the Symfony serializer
symfony/service-contracts          v1.1.5  v1.1.5  Generic abstractions related to writing services
symfony/stopwatch                  v4.3.3  v4.3.3  Symfony Stopwatch Component
symfony/templating                 v4.3.3  v4.3.3  Symfony Templating Component
symfony/test-pack                  v1.0.6  v1.0.6  A pack for functional and end-to-end testing within a Symfony app
symfony/translation                v4.3.3  v4.3.3  Symfony Translation Component
symfony/translation-contracts      v1.1.5  v1.1.5  Generic abstractions related to translation
symfony/twig-bridge                v4.3.3  v4.3.3  Symfony Twig Bridge
symfony/twig-bundle                v4.3.3  v4.3.3  Symfony TwigBundle
symfony/validator                  v4.3.3  v4.3.3  Symfony Validator Component
symfony/var-dumper                 v4.3.3  v4.3.3  Symfony mechanism for exploring and dumping PHP variables
symfony/var-exporter               v4.3.3  v4.3.3  A blend of var_export() + serialize() to turn any serializable...
symfony/web-link                   v4.3.3  v4.3.3  Symfony WebLink Component
symfony/web-profiler-bundle        v4.3.3  v4.3.3  Symfony WebProfilerBundle
symfony/web-server-bundle          v4.3.3  v4.3.3  Symfony WebServerBundle
symfony/webpack-encore-bundle      v1.6.2  v1.6.2  Integration with your Symfony app & Webpack Encore!
symfony/workflow                   v4.3.3  v4.3.3  Symfony Workflow Component
symfony/yaml                       v4.3.3  v4.3.3  Symfony Yaml Component
```

#### PHP version

```
$ php -v
PHP 7.3.5 (cli) (built: May  4 2019 00:58:41) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.5, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.5, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.7.0beta1, Copyright (c) 2002-2018, by Derick Rethans
```

## Subject
I am getting following error when i clear doctrine cache:
```bash
[WARNING] Some commands could not be registered:



In BaseCommand.php line 33:

  Argument 2 passed to Sonata\BlockBundle\Command\DebugBlocksCommand::__construct() must be an instance of Sonata\
  BlockBundle\Block\BlockServiceManagerInterface, NULL given.

```

## Steps to reproduce
clear doctrine cache with `--no-debug` option:
```bash
bin/console doctrine:cache:clear-metadata --no-debug
bin/console doctrine:cache:clear-query --no-debug
bin/console doctrine:cache:clear-result --no-debug
```
It will work correctly without the `--no-debug` flag

### PR DESCRIPTION
**Pull request by stloyd**
This fixes issue when dependency injection was not providing required service but called default parent `__construct()` which has always `null` as second argument.

## Subject

I am targeting this branch, because this is bugfix.

Closes #624

## Changelog

```markdown
### Fixed

Prevent calling default `BaseCommand::__construct()` with default arguments, blocking extending commands from being properly registered. #624

### CODE REVIEW

File: tests/Command/DebugBlocksCommandTest.php [hunk 1]
```diff
@@ -73,30 +73,6 @@ public function testExecuteWithAlias(): void
         $this->assertSame("done!\n", $commandTester->getDisplay());
     }
 
-    /**
```
Comment (core23) [inline tests/Command/DebugBlocksCommandTest.php]: Why did you remove this?

**Comment by stloyd** [comment]
Closing this, seems like some strange bug in one of versions of HttpKernel and/or DependencyInjection that tried to initialize abstract class via reflection, which lead to fatal error.

Can't reproduce that on Symfony `^4.3.4` anymore.
